src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -82,7 +82,8 @@ public ServiceProviderEngineScope(ServiceProvider provider, bool isRootScope)
                 else
                 {
                     // sync over async, for the rare case that an object only implements IAsyncDisposable and may end up starving the thread pool.
-                    Task.Run(() => ((IAsyncDisposable)service).DisposeAsync().AsTask()).GetAwaiter().GetResult();
```

Comment:
Verified that the IL in this method always creates an instance of the closure class:
```
  IL_0000:  newobj     instance void Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope/'<>c__DisplayClass20_0'::.ctor()
```